src/rust/engine/src/nodes.rs

```diff
@@ -333,10 +333,10 @@ impl ExecuteProcess {
     process_config: externs::process::PyProcessConfigFromEnvironment,
   ) -> Result<Process, StoreError> {
     let env = externs::getattr_from_str_frozendict(value, "env");
-    let working_directory = match externs::getattr_as_optional_string(value, "working_directory") {
-      None => None,
-      Some(dir) => Some(RelativePath::new(dir)?),
-    };
+
+    let working_directory = externs::getattr_as_optional_string(value, "working_directory")
+      .map_err(|e| format!("Failed to get `working_directory` from field: {e}"))?
+      .and_then(|dir| RelativePath::new(dir).ok());
```

Comment:
@stuhood any suggestions here? I don't want to throw away the `RelativePath` error, but since we return a Result here, it needs a more complex error handling method. Is there a more ergonomic way that doesn't skip the error handling?